Sources/NotificationApi_Ios_Sdk/NotificationApi.swift

```diff
@@ -0,0 +1,94 @@
+import Foundation
+import UIKit
+
+public class NotificationApi: NSObject {
+    public static let shared = NotificationApi()
+
+    // MARK: - Private Members
+
+    fileprivate static let defaultConfig = NotificationApiConfig(baseUrl: "https://api.notificationapi.com")
+    fileprivate static let deviceInfo = NotificationApiDeviceInfo()
+    fileprivate static let authOptions: UNAuthorizationOptions = [.badge, .alert, .sound]
+
+    fileprivate var credentials: NotificationApiCredentials?
+    fileprivate var config: NotificationApiConfig?
+    fileprivate var restApi: NotificationApiRest?
+
+    // MARK: - Init & Configuration
+
+    private override init() {
+        super.init()
+    }
+
+    public func configure(withCredentials credentials: NotificationApiCredentials, withConfig config: NotificationApiConfig? = nil) {
+        self.credentials = credentials
+        self.config = config ?? NotificationApi.defaultConfig
+        self.restApi = NotificationApiRest(baseUrl: self.config!.baseUrl, credentials: self.credentials!, deviceInfo: NotificationApi.deviceInfo)
+    }
+
+    fileprivate func checkCredentialsAndAuthorization() async throws {
```

Comment:
This is a handy function that will check if all prerequisites are good before actually continuing with anything NotificationAPI related. 

This will also check to see if notification permissions has been requested yet: if not, then we print out a helpful clue to do so